src/journal/compress.c

```diff
@@ -41,6 +41,61 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(LZ4F_decompressionContext_t, LZ4F_freeDecompressionC
 #if HAVE_ZSTD
 DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_CCtx *, ZSTD_freeCCtx);
 DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_DCtx *, ZSTD_freeDCtx);
+
+static int zstd_ret_to_errno(size_t ret) {
+        switch (ZSTD_getErrorCode(ret)) {
+#if ZSTD_VERSION_NUMBER  >= 10301
+        case 70: /* ZSTD_error_dstSize_tooSmall */
+                return -ENOBUFS;
+        case 64: /* ZSTD_error_memory_allocation */
+                return -ENOMEM;
+#endif
+        default:
+                return -EBADMSG;
+        }
+}
+
+#  if ZSTD_VERSION_NUMBER < 10400
```

Comment:
weird spaces at beginning of line?

Reply:
do a `grep -r '#  if'` over the sources, the weird spaces are your coding guidelines?
(if anything, I did forget those on the conditional directly before)